### DISCUSSION TITLE
Complex mutations

### DISCUSSION THREAD

**Discussion by chrismcb** [mirumee/ariadne#887] [Q&A] [answered]
How do we do complex mutations?
For example, the documentation mentions:
```
input PollInput {
    question: String!,
    options: [PollOptionInput!]!
}

input PollOptionInput {
    label: String!
    color: String!
}
```
Lets assume we add:
```
type Mutation {
    createPollInput(input:PollInput!): PollPayload
}
```

So I assume we do something like:
```
@mutation.field("createPollInput")
def createPollInput(....):
   pass
```

What I want to do is have a resolver for PollOptionsInput as well
```
@something.field("options")
def resolve_poll_options(....):
   pass
```

The question, how do I make this work? What should "something" be? Do we do something similar for resolvers:
```
poll_input = ObjectType("createPollInput")
```

Or should me createPollInput function call reolve_poll_options directly?

**Comment by rafalp** [comment, marked as answer]
GraphQL doesn't support resolvers for Input's fields. Whatever logic you want to implement, you need to call it from mutation's resolver.
